benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -598,124 +599,16 @@ void GraphicsBenchmarkApp::Render()
-                    data.ambient                    = float4(0.3f);
-                    data.cameraViewProjectionMatrix = mCamera.GetViewProjectionMatrix();
-                    data.lightPosition              = float4(mLightPosition, 0.0f);
-                    data.eyePosition                = float4(mCamera.GetEyePosition(), 0.0f);
-                    mDrawCallUniformBuffers[i]->CopyFromSource(sizeof(data), &data);
-
-                    frame.cmd->PushGraphicsUniformBuffer(mSphere.pipelineInterface, /* binding = */ 0, /* set = */ 0, /* bufferOffset = */ 0, mDrawCallUniformBuffers[i]);
-                    frame.cmd->PushGraphicsSampledImage(mSphere.pipelineInterface, /* binding = */ 1, /* set = */ 0, mAlbedoTexture.sampledImageView);
-                    frame.cmd->PushGraphicsSampler(mSphere.pipelineInterface, /* binding = */ 2, /* set = */ 0, mAlbedoTexture.sampler);
-                    frame.cmd->PushGraphicsSampledImage(mSphere.pipelineInterface, /* binding = */ 3, /* set = */ 0, mNormalMapTexture.sampledImageView);
-                    frame.cmd->PushGraphicsSampler(mSphere.pipelineInterface, /* binding = */ 4, /* set = */ 0, mNormalMapTexture.sampler);
-                    frame.cmd->PushGraphicsSampledImage(mSphere.pipelineInterface, /* binding = */ 5, /* set = */ 0, mMetalRoughnessTexture.sampledImageView);
-                    frame.cmd->PushGraphicsSampler(mSphere.pipelineInterface, /* binding = */ 6, /* set = */ 0, mMetalRoughnessTexture.sampler);
-
-                    uint32_t indexCount = indicesPerDrawCall;
-                    // Add the remaining indices to the last drawcall
-                    if (i == currentDrawCallCount - 1) {
-                        indexCount += (currentSphereCount * mSphereIndexCount - currentDrawCallCount * indicesPerDrawCall);
-                    }
-                    uint32_t firstIndex = i * indicesPerDrawCall;
-                    frame.cmd->DrawIndexed(indexCount, /* instanceCount = */ 1, firstIndex);
-                }
-            }
-        }
-        frame.cmd->EndRenderPass();
-
-        // =====================================================================
-        // Fullscreen quads renderpasses
-        // =====================================================================
-        if (pFullscreenQuadsCount->GetValue() > 0) {
-            frame.cmd->BindGraphicsPipeline(mFullscreenQuads.pipeline);
-            frame.cmd->BindVertexBuffers(1, &mFullscreenQuads.vertexBuffer, &mFullscreenQuads.vertexBinding.GetStride());
-
-            for (size_t i = 0; i < pFullscreenQuadsCount->GetValue(); ++i) {
-                currentRenderPass = swapchain->GetRenderPass(imageIndex);
-                PPX_ASSERT_MSG(!currentRenderPass.IsNull(), "render pass object is null");
+        // Note: Transitions image layout PRESENT->RENDER before this first renderpass
+        RenderScene(swapchain, frame, imageIndex);
```

Comment:
I am a bit confused by some of the currentRenderPass logic inside Render* functions. Could you maybe pass just the renderpass to the Render* functions instead of swapchain + index? In the case of GUI you will pass the LOAD_OP_LOAD one. Since it looks like it only needs the renderpass from the swapchain.

Reply:
I see the point about `currentRenderPass` being confusing, I think it isn't the best idea to pass it as a parameter since `RenderFullscreenQuads()` may actually use multiple renderpasses. I've instead refactored the code again (see main comment for details)